src/handlers.cpp

```diff
@@ -1700,19 +1706,12 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
       _http_rc = HTTP_SERVER_UNAVAILABLE;
       break;
       // LCOV_EXCL_STOP
-    case 5001:
-    {
-      TRC_INFO("Server-Assignment answer with result code %d - reject", result_code);
-      SAS::Event event(this->trail(), SASEvent::REG_DATA_HSS_FAIL, 0);
-      SAS::report_event(event);
-      _http_rc = HTTP_NOT_FOUND;
-    }
-    break;
```

Comment:
I notice you've removed this branch. The code as it stands is catching a 5001 which is  `DIAMETER_AVP_UNSUPPORTED`. However the 3GPP experimental result code 5001 is `DIAMETER_ERROR_USER_UNKNOWN` which is what I think this code is trying to handle. This makes me think that a SAR can have an experimental result code which we're not logging. It also suggests the code is bugged because its handling the wrong result code field. Please can you:

* Check whether SARs can have an experimental result code. 
* If they can, log it to SAS and display it. 
* Raise an issue covering the fact that this code looks wrong.